lib/data/multipleBackendGateway.js

```diff
@@ -31,6 +69,37 @@ const multipleBackendGateway = {
                 callback(errors.InternalError);
             });
         }
+        // client is AWS SDK
+        if (client.clientType === 'aws_s3') {
+            const awsKey = _createAwsKey(keyContext.bucketName,
+               keyContext.objectKey, client.bucketMatch);
+            // TODO: if object to be encrypted, use encryption
+            // on AWS
+            return client.putObject({
```

Comment:
Why not create a client wrapper for AWS, so that the code gets out of the way of multiple backend management?